File: baekjoon/dp/13398.py
import sys
from typing import List

input = sys.stdin.readline


def solution(nums: List[int]) -> int:
    dp = [[0] * len(nums) for _ in range(2)]
    dp[0][0] = nums[0]

    answer = nums[0]
    for i in range(1, len(nums)):
        dp[0][i] = max(dp[0][i - 1] + nums[i], nums[i])
        dp[1][i] = max(dp[1][i - 1] + nums[i], dp[0][i - 1])
        answer = max(answer, max(dp[0][i], dp[1][i]))
    print(answer)


if __name__ == "__main__":
    _ = input()
    nums = list(map(int, input().split()))
    solution(nums)

File: baekjoon/dp/13398.js
function solution(length, numbers) {
  const dp = Array(length)
    .fill(null)
    .map(() => Array(2).fill(Number.MIN_SAFE_INTEGER));

  dp[0][0] = numbers[0];

  for (let i = 1; i < length; i++) {
    dp[i][0] = Math.max(dp[i - 1][0] + numbers[i], numbers[i]);
    dp[i][1] = Math.max(dp[i - 1][0], dp[i - 1][1] + numbers[i]);
  }

  return Math.max(...dp.map(row => Math.max(...row)));
}

function main() {
  const fs = require('fs');
  const input = fs
    .readFileSync('./baekjoon/dp/13398.txt')
    .toString()
    .trim()
    .split('\n')
    .map(v => v.split(' ').map(Number));

  console.log(solution(input[0][0], input[1]));
}

if (require.main === module) {
  main();
}
